tests/unit/getUserByAccessKey.js

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable operator-linebreak */
+'use strict'; // eslint-disable-line
+
+const assert = require('assert');
+const http = require('http');
+const { errors } = require('arsenal');
+const querystring = require('querystring');
+const IAMClient = require('../../lib/IAMClient');
+
+
+const testUserId = 'testUserId';
+const testAccessKey = 'testAccessKey';
+const testSecretKey = 'testSecretKey';
+const notExistingAccessKey = 'notExistingAccessKey';
+
+const serverDB = {
+    [testAccessKey]: {
+        id: testAccessKey,
+        value: testSecretKey,
+        createDate: '2022-08-11T17:57:00Z',
+        status: 'Active',
+        userId: testUserId,
+    },
+    [testUserId]: {
+        arn: 'arn:aws:iam::405435207934:user/bart',
+        id: testUserId,
+        emailAddress: '',
+        name: 'bart',
+        createDate: '2022-08-11T17:57:00Z',
+        parentId: '405435207934',
+    },
+};
+
+function handler(req, res) {
+    const index = req.url.indexOf('?');
+    const data = querystring.parse(req.url.substring(index + 1));
+
+    let output = null;
+    try {
+        res.writeHead(200);
+        output = JSON.stringify(serverDB[serverDB[data.accessKey].userId]);
```

Comment:
yeah!  I do not understand this either.

Reply:
this is nested DB calls 😅 , first use data.accessKey as a key to search in DB, `serverDB[data.accessKey]` is the accessKey instance, from the accessKey instance, we get its userId, then use this userId as a key to search in DB to get the User instance

Reply:
It might be better to add a comment or use variables with appropriate names to make this code easier to read. What do you think? :slightly_smiling_face: I could understand, but it should be easier to read IMO :smile:

Reply:
yes! make sense

Reply:
https://github.com/scality/vaultclient/pull/406/commits/5cc7848d014b0349dc3606d6f68f76fc6fd1023c